### PR TITLE
Add documentions, gitignore, and deps script

### PR DESCRIPTION
**Pull request by sfricke-samsung**
Low hanging fruit of #32 

- Add basic documentation (to be expanded on as needed)
- .gitignore and .clang-format files from Validation Layers
- update dep script from glslang/validation layers to properly get Vulkan Headers to ensure building works

### CODE REVIEW

**Comment by sfricke-samsung** [comment]
@djdeath are you in charge of merging things in? As in who to ask to review things

**Comment by djdeath** [comment]
> @djdeath are you in charge of merging things in? As in who to ask to review things

Pretty much. But it won't be long before you're more qualified than I am ;)

**Comment by djdeath** [comment]
Maybe you should also update the .travis.yml file so that it uses the `scripts/update_deps.py` script to pull in the deps.

**Comment by sfricke-samsung** [comment]
Good call with the CI suggestion was going to try to have CI changes be its own MR, but seems to be in this initial PR scope to sync the headers everywhere

File: .travis.yml [hunk 1]
```diff
@@ -35,23 +38,44 @@ matrix:
...
+  - set +e
 
-  - cmake .
-    -DVulkanHeaders_INCLUDE_DIR=$PWD/Vulkan-Headers-$VULKAN_HEADERS_VERSION/include
-    -DVulkanRegistry_DIR=$PWD/Vulkan-Headers-$VULKAN_HEADERS_VERSION/registry
-    || travis_terminate 1
+script:
+  - set -e
+  - |
+    if [ $TRAVIS_OS_NAME = linux ]; then
```
Comment (djdeath) [inline .travis.yml:67]: This seems to skip the build on MacOS.

```
if [[ "$VULKAN_BUILD_TARGET" == "LINUX" ]] || [[ "$VULKAN_BUILD_TARGET" == "MACOS" ]]; then
```

Comment (sfricke-samsung) [reply .travis.yml]: Good catch, restructured file

Comment (djdeath) [reply .travis.yml]: Really sorry, I updated my comment after realizing I used the wrong variable name : 

```
if [ $TRAVIS_OS_NAME = linux -o $TRAVIS_OS_NAME = osx ]; then
```

**Comment by djdeath** [comment]
All good, ship it!

**Comment by sfricke-samsung** [comment]
Ya, so validation layers don't build on macos, I don't have a Mac, so sorry for the extreme game of "What is supported in zch but not bash via guess and check"

**Pull request merged**